charts/kargo/templates/controller/service-account.yaml

```diff
@@ -7,4 +7,10 @@ metadata:
   labels:
     {{- include "kargo.labels" . | nindent 4 }}
     {{- include "kargo.controller.labels" . | nindent 4 }}
+  {{- with (mergeOverwrite (deepCopy .Values.global.annotations) .Values.controller.annotations) }}
+  annotations:
+    {{- range $key, $value := . }}
+      {{ $key }}: {{ $value | quote }}
+      {{- end }}
+  {{- end }}
```

Comment:
I think this is maybe not how I would approach this. Thus far, this whole strategy of merging global labels and/or annotations with component-specific ones has only been used on workloads (Pods) and things managing workloads (Deployments and CronJobs). Mostly this has been done for the benefit of people who, due to policy or whatever, need workloads labeled or annotated a certain way.

If we approach things in this way, someone is going to add IRSA-related annotations to either `.Values.global.annotations` or `.Values.controller.annotations`, and depending on which, they're going to end up with those annotations added in a whole lot of places where they are not needed -- in which case they are just noise.

I think the best approach might be to add a dedicated `controller.irsa.roleName` or something of that nature and use its value here if and only if non-empty defined.

Reply:
Yeah, that was actually my first thought 😄 I went with this approach instead, since it seems like there was a convention of doing that thing above... I'll amend this to do something like `controller.serviceAccount.annotations` instead.

Reply:
I added a dedicated annotations field for the service account, should this be something which should be merged with `.Values.global.annotations`?

Reply:
For now I think no. We can revisit if we determine there is a need.